dali/operators/util/get_property.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_UTIL_GET_PROPERTY_H_
+#define DALI_OPERATORS_UTIL_GET_PROPERTY_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "dali/operators/util/property.h"
+#include "dali/pipeline/data/type_traits.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+
+template <typename Backend>
+class GetProperty : public Operator<Backend> {
+ public:
+  explicit GetProperty(const OpSpec &spec) : Operator<Backend>(spec) {
+    auto property_name = spec.template GetArgument<std::string>("key");
+    PropertyFactory(property_name);
+  }
+
+  ~GetProperty() override = default;
+  DISABLE_COPY_MOVE_ASSIGN(GetProperty);
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
+    const auto &input = ws.template Input<Backend>(0);
+    output_desc.resize(1);
+    output_desc[0].shape = property_->GetShape(input);
+    output_desc[0].type = property_->GetType(input);
+    return true;
+  }
+
+  void RunImpl(workspace_t<Backend> &ws) override {
+    property_->FillOutput(ws);
+  }
+
+ private:
+  void PropertyFactory(const std::string &property_key) {
```

Comment:
Went with `std::unique_ptr<tensor_property::Property<Backend>> PropertyFactory(...)` since it does look more like a good practice to initialize members in the ctor initialize list